modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -531,6 +531,35 @@ class MishSubgraph : public Subgraph
     }
 };
 
+// softplus(x) = log(exp(x) + 1)
+class SoftplusSubgraph: public Subgraph
+{
+public:
+    SoftplusSubgraph()
+    {
+        int input = addNodeToMatch("");
+        int exp = addNodeToMatch("Exp", input);
+        int addVal = addNodeToMatch("");
+        int add = addNodeToMatch("Add", addVal, exp);
+        addNodeToMatch("Log", add);
+        setFusedNode("Softplus", input);
+    }
+};
+
+class SoftplusSubgraph2: public Subgraph
+{
+public:
+    SoftplusSubgraph2()
+    {
+        int input = addNodeToMatch("");
+        int exp = addNodeToMatch("Exp", input);
+        int addVal = addNodeToMatch("");
+        int add = addNodeToMatch("Add", exp, addVal);
+        addNodeToMatch("Log", add);
+        setFusedNode("Softplus", input);
+    }
+};
```

Comment:
We have two very similar `SoftplusSubgrap` because `Softplus` contains an `Add` which has two inputs (one is variable and one is `Constant` 1). Since the order of these two inputs is random, we use two `SoftplusSubgraph` to handle this. Is there a better solution?

Reply:
I think that simplifier should just match different nodes. You can make a small onnx model of both cases, remove the second class and test if it's working.

Reply:
Thanks for code reviewing! Actually, I have verified it. And the `SoftplusSubgraph` can work on [Yolov4](https://drive.google.com/file/d/1WBZO-YCqYTO73j8w_soMPD3ljp8fAD_-/view?usp=sharing), and the `SoftplusSubgraph2` can only work on [this test case](https://github.com/opencv/opencv_extra/pull/990).

Reply:
Hm, actually it makes sense - the operation is not required to be commutative. You could use CRTP to statically parametrise the order, but I don't think the complexity is worth it in this case. The second solution is to add logic of reordering into subgraph matching, but it'll take a while. I think this works for now.